cypress/integration/email/mention_email_notification_spec.js

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/* eslint max-nested-callbacks: ["error", 4] */
+
+import users from '../../fixtures/users.json';
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+const reUrl = /(https?:\/\/[^ ]*)/;
+
+const user1 = users['user-1'];
+const user2 = users['user-2'];
+
+const text = `Hello @${user2.username}`;
+const feedbackEmail = 'test@example.com';
+
+describe('Email notification', () => {
+    before(() => {
+        cy.apiUpdateConfig({EmailSettings: {FeedbackEmail: feedbackEmail}});
+    });
+
+    it('post a message that mentions a user', () => {
+        cy.apiLogin('user-1');
+        cy.visit('/ad-1/channels/town-square');
+        cy.postMessage(`${text}{enter}`);
+
+        // Wait for a while to ensure that email notification is sent.
+        cy.wait(TIMEOUTS.SMALL);
+
+        cy.task('getRecentEmail', {username: 'user-2'}).then((response) => {
+            verifyEmailNotification(response, 'eligendi', 'Town Square', user2, user1, text, feedbackEmail);
+
+            const bodyText = response.data.body.text.split('\n');
+
+            const permalink = bodyText[9].match(reUrl)[0];
+            const permalinkPostId = permalink.split('/')[5];
+            cy.visit(permalink);
+
+            const postText = `#postMessageText_${permalinkPostId}`;
+            cy.get(postText).should('have.text', text);
+
+            cy.getLastPostId().then((postId) => {
+                // * Should match last post and permalink post IDs
+                expect(permalinkPostId).to.equal(postId);
+            });
+        });
+    });
+});
+
+function verifyEmailNotification(response, teamDisplayName, channelDisplayName, mentionedUser, byUser, message, fromEmail) {
+    const isoDate = new Date().toISOString().substring(0, 10);
+    const {data, status} = response;
+
+    // * Should return success status
+    expect(status).to.equal(200);
+
+    // * Verify that email is addressed to user-2
+    expect(data.to[0]).to.contain(mentionedUser.email);
+
+    // * Verify that email is from default feedback email
+    expect(data.from).to.contain(fromEmail);
+
+    // * Verify that date is current
+    expect(data.date).to.contain(isoDate);
+
+    // * Verify that the email subject is correct
+    expect(data.subject).to.contain(`[Mattermost] Notification in ${teamDisplayName}`);
+
+    // * Verify that the email body is correct
+    const bodyText = data.body.text.split('\n');
+    expect(bodyText.length).to.equal(16);
+    expect(bodyText[1]).to.contain('You have a new notification.');
```

Comment:
Should we do `expect(foo).to.equal('bar')` vs `contain`? Contain could still pass if there is more text than what we expected, although depending on the text, there may be extraneous stuff we don't care about.

Reply:
This applies to all `contains` in this file, btw.

Reply:
Good point.  Apply to those which are expected to have exact value.